Lacrit/assignment2/binary_tree.hpp

```diff
@@ -0,0 +1,85 @@
+#include <vector>
+template <class T>
+class Btree;
+
+template <class T>
+struct Node
+{
+    private: 
+		T data;
+		Node *left; 
+		Node *right;
+		friend class Btree<T>;
+};
+
+template <class T>
+class Btree 
+{
+	// I need to have a reference to a pointer, otherwise it would overide the pointer? (Please, correct me if I`m wrong here)
```

Comment:
That is an option, but it's a bit confusing. 
I think is better to have a pointer to a pointer.
In any case, it will be better if there was a way to construct any tree you can think of, this can only create unbalanced trees, right?